include/gsl/gsl-lite.hpp

```diff
@@ -704,9 +704,11 @@
 #endif
 
 #if !gsl_HAVE( NOEXCEPT ) || defined( gsl_TESTING_ )
-# define gsl_noexcept /*noexcept*/
+# define gsl_noexcept           /*noexcept*/
+# define gsl_noexcept_iff(expr) /*noexcept(noexcept(expr))*/
```

Comment:
The idea behind the naming was that the function would be `noexcept` if and only if the expression was `noexcept` as well, but I'm not particularly happy with the name, either. I wanted to avoid having nested macros `gsl_noexcept_if( gsl_noexcept( ... ) )`, but if you prefer that, I'll write it that way.

Reply:
A function/macro named "*_if" or "*_iff" that *doesn't* expect a Boolean argument feels weird to me.

For a Boolean argument "A", the "if and only if" is redundant: we are *defining* a Boolean property "B" of the function (stating whether the function is `noexcept`). It is true that "B" can be defined implicitly by saying "(A ⇒ B) ∧ (¬A ⇒ ¬B)", but why would we if we can simply say "B := A"?